pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -0,0 +1,625 @@
+    // Status of the container.
+    optional ContainerState state = 3;
+    // Creation time of the container.
+    optional int64 created_at = 4;
+    // Start time of the container.
+    optional int64 started_at = 5;
+    // Finish time of the container.
+    optional int64 finished_at = 6;
+    // Exit code of the container.
+    optional int32 exit_code = 7;
+    // Reference to the image in use. For most runtimes, this should be an
+    // image ID
+    optional string image_ref = 8; 
+    // A string explains why container is in such a status.
+    optional string reason = 9;
+    // Labels are key value pairs that may be used to scope and select individual resources.
+    map<string,string> labels = 10;
+    // Annotations is an unstructured key value map.
+    map<string,string> annotations = 11;
+    // Mounts specifies mounts for the container
+    repeated Mount mounts = 12;
+}
+
+message ContainerStatusResponse {
+    // The status of the container
+    optional ContainerStatus status = 1;
+}
+
+message ExecRequest {
+    // The id of the container
+    optional string container_id = 1;
+    // The cmd to execute
+    repeated string cmd = 2;
+    // Whether use tty
+    optional bool tty = 3;
+    // Streaming stdin
+    optional bytes stdin = 4;
+}
+
+message ExecResponse {
+    // Streaming stdout
+    optional bytes stdout = 1;
+    // Streaming stderr
+    optional bytes stderr = 2;
+}
+
+message ImageFilter {
+    // The spec of the image
+    optional ImageSpec image = 1;
+}
+
+message ListImagesRequest {
+    // The filter to list images
+    optional ImageFilter filter = 1;
+}
+
+// Basic information about a container image.
+message Image {
+    // ID of the image.
+    optional string id = 1;
+    // Other names by which this image is known.
+    repeated string repo_tags = 2;
+    // Digests by which this image is known.
+    repeated string repo_digests = 3;
+    // The size of the image in bytes.
+    optional uint64 size = 4;
+}
+
+message ListImagesResponse {
+    // List of images
+    repeated Image images = 1;
+}
+
+message ImageStatusRequest {
+    // The spec of the image
+    optional ImageSpec image = 1;
+}
+
+message ImageStatusResponse {
+    // The status of the image
+    optional Image image = 1;
+}
+
+// AuthConfig contains authorization information for connecting to a registry.
+message AuthConfig {
+    optional string username = 1;
+    optional string password = 2;
+    optional string auth = 3;
+    optional string server_address = 4;
+    // IdentityToken is used to authenticate the user and get
+    // an access token for the registry.
+    optional string identity_token = 5;
+    // RegistryToken is a bearer token to be sent to a registry
+    optional string registry_token = 6;
+}
+
+message PullImageRequest {
+    // The spec of the image
+    optional ImageSpec image = 1;
+    // The auth config for pulling image
+    optional AuthConfig auth = 2;
+}
+
+message PullImageResponse {}
```

Comment:
@feiskyer I guess I have a broader question here: Suppose we use some neutral pluggable image puller (e.g. skopeo) that discovers and downloads the image into disk. Then how the the runtime knows where to find the on disk image if it is not given a reference (such as image id, or file path)? Or do I miss anything?